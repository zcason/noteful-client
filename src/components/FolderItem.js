import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import DefaultContext from './context/DefaultContext';
import PropTypes from 'prop-types';

class FolderItem extends Component {
    static contextType = DefaultContext;

    deleteNote = (id) => {
        fetch(`${this.context.url}/notes/${id}`, {
            method: 'DELETE',
            headers: {
              'content-type': 'application/json'
            },
          })
          .then( r=>{
            this.context.updateStore();
          })
          .catch( e => {
              throw new Error(`Error deleting note: ${e.message}`);
          } );;
    }
    deleteFolder = (id) => {
        fetch(`${this.context.url}/folders/${id}`, {
            method: 'DELETE',
            headers: {
              'content-type': 'application/json'
            },
        })
        .then( r => {
            this.context.updateStore();
            this.props.history.push('/');
        } )
        .catch( e => {
            throw new Error(`Error deleting folder: ${e.message}`);
        } );;
    }
    getRidOfMe = () => {
        // remove all notes with this folder id
        this.props.store.notes.forEach( note => {
            if ( note.folderId === this.props.id ) this.deleteNote( note.id );
        } );
        // remove this folder
        this.deleteFolder( this.props.id );
    }

    render(){
        const { id, name, history } = this.props;

        return (
            <li key={id} className={`folder--item ${history.location.pathname.includes(id) ? 'folder--item--checked' : ''}`}>
                <Link 
                    to={`/folder/${id}`}
                >
                    {name}
                </Link>
                <button 
                className="folder--item--delete"
                onClick={()=>this.getRidOfMe()}>delete</button>
            </li>
       );
    }
}

FolderItem.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    store: PropTypes.object.isRequired
}

export default withRouter(FolderItem);
