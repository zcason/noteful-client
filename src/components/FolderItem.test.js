import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import store from '../dummy-store';
import FolderItem from './FolderItem';

describe(`FolderItem component`, () => {
    const props = {
        id: 'testid',
        name: 'testname',
        store: store
    }
  
    it('renders FolderItem by default', () => {
      const wrapper = shallow(<FolderItem />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders fodler with name testname when in props', () => {
      const component = shallow(<FolderItem {...props} />)
        .find('testname')
      expect(toJson(component)).toMatchSnapshot()
    })
  })
  