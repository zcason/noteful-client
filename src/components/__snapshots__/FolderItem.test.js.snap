// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderItem component renders FolderItem by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`FolderItem component renders fodler with name testname when in props 1`] = `null`;
